### PR TITLE
fix: search blur state for every single charater input

### PR DESCRIPTION
**Pull request by leangseu-edx**
**TL;DR -** fix search bar to remain focus on every keyboard input

JIRA: [AU-723](https://2u-internal.atlassian.net/browse/AU-723)

**What changed?**

- Instead of using `onChange`, we use `onBlur` and `onSubmit` to accommodate the use case. I also tested with `this.setState` but was unsuccessful
- NOTE: This approach was trying to preserve the behavior/design choice for the interaction between **search bar** and **filter side panel**

**Developer Checklist**
- [x] Test suites passing
- [ ] Documentation and test plan updated, if applicable
- [x] Received code-owner approving review
- [ ] Bumped version number [package.json](../package.json)

**Testing Instructions**

[ How should a reviewer test this PR? ]

**Reviewer Checklist**

Collectively, these should be completed by reviewers of this PR:

- [ ] I've done a visual code review
- [ ] I've tested the new functionality

FYI: @openedx/content-aurora

### CODE REVIEW

**Review by mattcarter — APPROVED** [review]

**Review by muselesscreator — APPROVED** [review]
LGTM. Had a feeling it was something like this. :-)
